File: MusicWeb.Server/MusicWeb.Models/Entities/Users/UserFavoriteAlbum.cs
using MusicWeb.Models.Entities.Base;
using MusicWeb.Models.Identity;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MusicWeb.Models.Entities
{
    public class UserFavoriteAlbum : BaseEntity
    {
        public DateTime FavoriteDate { get; set; }

        public string UserId { get; set; }
        public int AlbumId { get; set; }
        public virtual Album Album { get; set; }
        public virtual ApplicationUser User { get; set; }
    }
}

File: MusicWeb.Client/src/models/UserFavoriteAlbum.js
export default class UserFavoriteAlbum {
  id = null;
  userId = null;
  albumId = null;
  album = {};
  user = {};
  albumId = null;
  constructor($data) {
    if ($data) {
      this.id = $data.id;
      this.userId = $data.userId;
      this.albumId = $data.albumId;
      this.album = $data.album ? $data.album : {};
      this.user = $data.user ? $data.user : {};
    }
  }
}
